method-info/thousand-pound-correction/thousand_pound_correction.md

```diff
@@ -0,0 +1,153 @@
+
+```python
+def thousand_pounds(
+    unique_identifier: Optional[str],  # Unique identifier e.g. a question
+    # code/ruref/period/id/combination of all of these
+    principal_variable: float,  # Original response value provided for
+    # the 'current' period
+    predictive: Optional[float],  # Value used for 'previous'
+    # response (Returned/Imputed/Constructed)
+    auxiliary: Optional[float],  # Calculated response for the
+    # 'previous' period
+    upper_limit: float,  # Upper bound of 'error value' threshold
+    lower_limit: float,  # Lower bound of 'error value' threshold
+    target_variables: List[TargetVariable], # identifier/value pairs
+    precision: Optional[int],  # Precision is used by the decimal
+    # package to ensure a specified accuracy
+    # used throughout method processing
+)
+```
+
+Calling the method with the example data:
+```python
+from thousand_pounds import thousand_pounds
+
+output = thousand_pounds(
+    unique_identifier = "12340000001-201409-q100",
+    principal_variable = "50000000",
+    predictive = "60000",
+    auxiliary = "30000",
+    upper_limit = "1350",
+    lower_limit = "350",
+    target_variables = {"q101": 500,
+                        "q102": 1000,
+                        "q103": 1500,
+                        "q104": None},
+    precision = 2
+)
+```
+The output will be returned as an object. You will need to destructure the object to extract the values and one way of doing this is using the built-in function vars(). vars() is used to return the \_\_dict\_\_ attribute for the specified module, class, instance or any other object with a \_\_dict\_\_ attribute.
+```python
+print(vars(output))
+```
+
+```bash
+{'unique_identifier': '12340000001-201409-q100', 'principal_final_value': '5.0E+4', 'target_variables': [TargetVariable(identifier='q101', original_value='500', final_value='0.5'), TargetVariable(identifier='q102', original_value='1000', final_value='1'), TargetVariable(identifier='q103', original_value='1500', final_value='1.5'), TargetVariable(identifier='q104', original_value=None, final_value=None)], 'tpc_ratio': '8.3E+2', 'tpc_marker': 'C'}
+
+```
+
+Example output table:
+
+| unique_identifier | principal_final_value | target_variables | tpc_ratio | tpc_marker |
+| --- | --- | --- | --- | --- |
+| '12340000001-201409-q100' | '5.0E+4' | [TargetVariable(identifier='q101', original_value='500', final_value='0.5'), TargetVariable(identifier='q102', original_value='1000', final_value='1'), TargetVariable(identifier='q103', original_value='1500', final_value='1.5'), TargetVariable(identifier='q104', original_value=None, final_value=None)] | '8.3E+2' | 'C' |
+
+Output attributes:
+* unique_identifier - Unique identifier e.g. a question code/ruref/period/id/combination of all of these
+* principal_final_value – Output value that may or may not be adjusted
+* target_variables - adjusted identifier/value pairs (returned as TargetVariable objects)
```

Comment:
target variables may not be adjusted, same as principal

Reply:
Made the changes, thanks!